src/NovelRT.Interop/Animation/NovelRTSpriteAnimatorState.cpp

```diff
@@ -0,0 +1,201 @@
+    }
+    catch (const std::exception& ex) {
+        if (errorMessage != nullptr) {
+            *errorMessage = ex.what();
+        }
+        return NOVELRT_FAILURE;
+    }
+    
+    return NOVELRT_SUCCESS;    
+}
+
+//std::vector<SpriteAnimatorFrame>
+NovelRTSpriteAnimatorFrameVector NovelRT_SpriteAnimatorFrameVector_create() {
+    std::vector<Animation::SpriteAnimatorFrame>* vector = new std::vector<Animation::SpriteAnimatorFrame>();
+    return reinterpret_cast<NovelRTSpriteAnimatorFrameVector>(vector);
+}
+
+int32_t NovelRT_SpriteAnimatorFrameVector_addFrame(NovelRTSpriteAnimatorFrameVector vector, NovelRTSpriteAnimatorFrame frame, const char** errorMessage) {
+    if (vector == nullptr || frame == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    std::vector<Animation::SpriteAnimatorFrame> cppVector = *reinterpret_cast<std::vector<Animation::SpriteAnimatorFrame>*>(vector);
+    Animation::SpriteAnimatorFrame cppFrame = *reinterpret_cast<Animation::SpriteAnimatorFrame*>(frame);
+    cppVector.push_back(cppFrame);
+    return NOVELRT_SUCCESS;
+}
+
+int32_t NovelRT_SpriteAnimatorFrameVector_getFrameAtIndex(NovelRTSpriteAnimatorFrameVector vector, int32_t index, NovelRTSpriteAnimatorFrame* outputFrame, const char** errorMessage) {
+    if (vector == nullptr || outputFrame == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    std::vector<Animation::SpriteAnimatorFrame> cppVector = *reinterpret_cast<std::vector<Animation::SpriteAnimatorFrame>*>(vector);
+    Animation::SpriteAnimatorFrame* cppFrame = new Animation::SpriteAnimatorFrame();
+    *cppFrame = cppVector.at(index);
+    return NOVELRT_SUCCESS;
+}
+
+int32_t NovelRT_SpriteAnimatorFrameVector_removeFrameAtIndex(NovelRTSpriteAnimatorFrameVector vector, int32_t index, const char** errorMessage) {
+    if (vector == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    std::vector<Animation::SpriteAnimatorFrame> cppVector = *reinterpret_cast<std::vector<Animation::SpriteAnimatorFrame>*>(vector);
+    cppVector.erase(cppVector.begin() + index);
+    return NOVELRT_SUCCESS;
+}
+
+int32_t NovelRT_SpriteAnimatorFrameVector_delete(NovelRTSpriteAnimatorFrameVector vector, const char** errorMessage) {
+    if (vector == nullptr) {
+        if (errorMessage != nullptr) {
+            *errorMessage = NovelRT_getErrMsgIsNullptr();
+        }
+        return NOVELRT_FAILURE;
+    }
+
+    std::vector<Animation::SpriteAnimatorFrame>* cppVector = reinterpret_cast<std::vector<Animation::SpriteAnimatorFrame>*>(vector);
+    cppVector->clear();
+    delete cppVector;
```

Comment:
Can you clarify what's going on here?

Reply:
There's no need for delete here - Removed.